src/main/java/org/jabref/gui/JabRefGUI.java

```diff
@@ -253,12 +252,13 @@ public void onHiding(WindowEvent event) {
 
     private void saveWindowState() {
         GuiPreferences preferences = preferencesService.getGuiPreferences();
-        preferences.setPositionX(mainStage.getX());
-        preferences.setPositionY(mainStage.getY());
-        preferences.setSizeX(mainStage.getWidth());
-        preferences.setSizeY(mainStage.getHeight());
+        if (!mainStage.isMaximized()) {
+            preferences.setPositionX(mainStage.getX());
+            preferences.setPositionY(mainStage.getY());
+            preferences.setSizeX(mainStage.getWidth());
+            preferences.setSizeY(mainStage.getHeight());
+        }
         preferences.setWindowMaximised(mainStage.isMaximized());
```

Comment:
I would drop this line, too.

This should keep the fix at https://github.com/JabRef/jabref/issues/4939